src/BaseBlockTracker.ts

```diff
@@ -137,7 +139,9 @@ export class BaseBlockTracker extends SafeEventEmitter {
     }
     this._isRunning = false;
     this._setupBlockResetTimeout();
-    this._end();
+    this._end().then(() => {
```

Comment:
Better to convert this to `async/await` and use `await` here so that the stack trace is preserved.